hack/test/e2e-platform.sh

```diff
@@ -0,0 +1,65 @@
+source ./hack/test/e2e-runner.sh
+
+## Download kustomize and template out capi cluster, then deploy it
```

Comment:
Broke this out into its own script, as it was a fair amount of logic. This will create the cluster in a given `$PLATFORM`, which will be gce/azure for now.